pkg/network/node/sdn_controller.go

```diff
@@ -167,15 +168,35 @@ func (plugin *OsdnNode) SetupSDN() (bool, error) {
 	}
 
 	gwCIDR := fmt.Sprintf("%s/%d", localSubnetGateway, localSubnetMaskLength)
+
+	if err := waitForOVS("unix", "/var/run/openvswitch/db.sock"); err != nil {
+		return false, err
+	}
+
+	var changed bool
 	if plugin.alreadySetUp(gwCIDR, clusterNetworkCIDRs) {
 		glog.V(5).Infof("[SDN setup] no SDN setup required")
-		return false, nil
+	} else {
+		glog.Infof("[SDN setup] full SDN setup required")
+		if err := plugin.setup(clusterNetworkCIDRs, localSubnetCIDR, localSubnetGateway, gwCIDR); err != nil {
+			return false, err
+		}
+		changed = true
 	}
-	glog.V(5).Infof("[SDN setup] full SDN setup required")
 
-	err = plugin.oc.SetupOVS(clusterNetworkCIDRs, serviceNetworkCIDR, localSubnetCIDR, localSubnetGateway)
-	if err != nil {
-		return false, err
+	// TODO: make it possible to safely reestablish node configuration after restart
+	// If OVS goes down and fails the health check, restart the entire process
+	healthFn := func() bool { return plugin.alreadySetUp(gwCIDR, clusterNetworkCIDRs) }
+	runOVSHealthCheck("unix", "/var/run/openvswitch/db.sock", healthFn)
```

Comment:
Maybe make `"unix"` and `"/var/run/openvswitch/db.sock"` be constants in healthcheck.go

Reply:
k